design/storage-class-device-set.md

```diff
@@ -0,0 +1,493 @@
+# Rook StorageClassDeviceSets
+
+**Target version**: Rook 1.1
+
+## Background
+
+The primary motivation for this feature is to take advantage of the mobility of
+storage in cloud-based environments by defining storage based on sets of devices
+consumed as block-mode PVCs. In environments like AWS you can request remote
+storage that can be dynamically provisioned and attached to any node in a given
+Availability Zone (AZ). The design can also accommodate non-cloud environments
+through the use of local PVs.
+
+## StorageClassDeviceSet struct
+
+```go
+// File: pkg/apis/ceph.rook.io/v1/types.go
+package v1
+
+import (
+    v1 "k8s.io/api/core/v1"
+
+    rook "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+)
+
+type StorageClassDeviceSets []StorageClassDeviceSet
+
+struct StorageClassDeviceSet {
+    Name                 string // A unique identifier for the set
+    Count                int // Number of devices in this set
+    Resources            v1.ResourceRequirements // Requests/limits for the devices
+    Placement            rook.Placement // Placement constraints for the devices
+    Config               map[string]string // Provider-specific device configuration
+    volumeClaimTemplates []v1.PersistentVolumeClaimSpec // List of PVC templates for the underlying storage devices
```

Comment:
whats the need of a list of PVCSpec template if we are having a count field in the StorageClassDeviceSet

Reply:
The count would apply to every template in the list, so if you have two templates and a count of 3 you would end up with 6 PVCs. This is intended to accommodate the scenario where one OSD is using multiple devices, and is shown in the last example in the doc.